convert/github/convert.go

```diff
@@ -107,23 +117,57 @@ func (d *Converter) ConvertFile(p string) ([]byte, error) {
 	return d.Convert(f)
 }
 
-// converts converts a github pipeline pipeline.
-func (d *Converter) convert() ([]byte, error) {
+// converts a GitHub pipeline to Harness pipeline.
+func (d *Converter) convert(ctx *context) ([]byte, error) {
 
 	// create the harness pipeline
 	pipeline := &harness.Pipeline{
 		Version: 1,
-		// Default: convertDefault(d.config),
+		Stages:  []*harness.Stage{},
 	}
 
-	// for _, steps := range d.config.Pipelines.Default {
-	// 	if steps.Stage != nil {
-	// 		// TODO support for fast-fail
-	// 		d.stage = steps.Stage // push the stage to the state
-	// 		stage := d.convertStage()
-	// 		pipeline.Stages = append(pipeline.Stages, stage)
-	// 	}
-	// }
+	for _, from := range ctx.pipeline {
+		if from == nil {
+			continue
+		}
+		pipeline.Name = from.Name
+
+		if from.Jobs != nil {
+			for name, job := range from.Jobs {
+				actionJob := &job
+				var cloneStage *harness.CloneStage
+				if actionJob != nil {
+					for _, step := range actionJob.Steps {
+						cloneStage = convertClone(step)
+						if cloneStage != nil {
+							break
+						}
+					}
+				}
+
+				pipeline.Stages = append(pipeline.Stages, &harness.Stage{
+					Name:     name,
+					Type:     "ci",
+					When:     convertCond(&from.On),
+					Strategy: convertStrategy(actionJob.Strategy),
+					Spec: &harness.StageCI{
+						Clone:    cloneStage,
+						Envs:     copyEnv(from.Environment),
+						Platform: convertRunsOn(actionJob.RunsOn),
+						Runtime: &harness.Runtime{
+							Type: "machine",
+							Spec: harness.RuntimeMachine{},
```

Comment:
we should use a pointer her.  `&harness.RuntimeCloud` or `&harness.RuntimeKuberentes`.  The machine runtime is for docker, which we are not currently supporting.  I would use cloud runtime for now